serde-generate/src/solidity.rs

```diff
@@ -783,21 +991,91 @@ function bcs_deserialize_offset_{name}(uint256 pos, bytes memory input) internal
                         let snake_name = named_format.name.to_snake_case();
                         let code_name = format.code_name();
                         let key_name = format.key_name();
-                        writeln!(out, "  {code_name}{data_location} {snake_name};")?;
-                        writeln!(out, "  if (choice == {idx}) {{")?;
-                        writeln!(out, "    (new_pos, {snake_name}) = bcs_deserialize_offset_{key_name}(new_pos, input);")?;
-                        writeln!(out, "  }}")?;
+                        writeln!(out, "    {code_name}{data_location} {snake_name};")?;
+                        writeln!(out, "    if (choice == {idx}) {{")?;
+                        writeln!(out, "        (new_pos, {snake_name}) = bcs_deserialize_offset_{key_name}(new_pos, input);")?;
+                        writeln!(out, "    }}")?;
                         entries.push(snake_name);
                     }
                 }
                 writeln!(
                     out,
-                    "  return (new_pos, {name}(choice, {}));",
+                    "    return (new_pos, {name}(choice, {}));",
                     entries.join(", ")
                 )?;
                 writeln!(out, "}}")?;
                 output_generic_bcs_deserialize(out, name, name, true)?;
             }
+            BytesN { size } => {
+                let name = format!("bytes{size}");
+                writeln!(out)?;
```

Comment:
Why so many `writeln!` instead of using `r#"` as in other places?

Reply:
Ok, changed.